### ISSUE TITLE
[Results] heading includes additional information

### ISSUE THREAD

**Issue by BobanL** [CDCgov/prime-simplereport#6392]
## Area or Component

Results Page

## Quick Summary

Header for results page includes information about how many results are avialble

## Accessibility Violation

[Heading not descriptive](https://docs.deque.com/issue-help/1.0.0/en/heading-not-descriptive)

## Action Requested

Do something similar to #5908

## Screenshots (optional)

![image](https://github.com/CDCgov/prime-simplereport/assets/10108172/b209818a-cb29-46be-94b2-056cc009bd66)
